src/cfn-resource-generator.ts

```diff
@@ -0,0 +1,144 @@
+import * as camelcase from 'camelcase';
+import * as j2j from 'json2jsii';
+
+export function sanitizeTypeName(typeName: string) {
+  const parts = typeName.split('::');
+  return parts.map(part => part.substr(0, 1).toUpperCase() + part.substr(1).toLowerCase()).join('');
+}
+
+export class CfnResourceGenerator {
+  sanitizedTypeName: string;
+
+  constructor(private readonly typeName: string, private readonly typeDef: AWS.CloudFormation.DescribeTypeOutput, private readonly schema: any) {
+    this.sanitizedTypeName = sanitizeTypeName(typeName);
+  }
+
+  public render(): string {
+    const code = new j2j.Code();
+    this.generateImports(code);
+    this.generateDefinitionTypes(code);
+    this.generateConstructClass(code);
+    return code.render();
+  }
+
+  private generateImports(code: j2j.Code) {
+    code.line("import * as cdk from '@aws-cdk/core';");
+    code.line();
+  }
+
+  private generateDefinitionTypes(code: j2j.Code) {
+    const gen = new j2j.TypeGenerator({
+      definitions: this.schema.definitions,
+    });
+
+    const schema = JSON.parse(JSON.stringify(this.schema));
+    this.schema.readOnlyProperties.map((prop: string) => prop.replace(/^\/properties\//, '')).forEach((prop: string) => {
+      delete schema.properties[prop];
+    });
+
+    gen.emitType(`Cfn${this.sanitizedTypeName}Props`, schema);
```

Comment:
I am assuming the name of this type is referenced in the constructor signature, so I would store all these names private properties (in the constructor of CfnResourceGenerator) and then use them everywhere